test/end-to-end/transactionType/transactionType.spec.js

```diff
@@ -0,0 +1,119 @@
+/* global element, by, browser */
+'use strict';
+
+const uuid   = require('node-uuid');
+const chai   = require('chai');
+const expect = chai.expect;
+
+const FU = require('../shared/FormUtils');
+const helpers = require('../shared/helpers');
+const components = require('../shared/components');
+
+helpers.configure(chai);
+
+describe('Transaction Types ::', () => {
+
+  // navigate to the page
+  before(() => helpers.navigate('#/admin/transaction_type'));
+
+  // after the creation the inserted id will be 11
+  let identifier = 11;
+
+  let newType = {
+    text: 'E2E Transaction Type',
+    description: 'Test transaction type',
+    type: 'income',
+    prefix: 'E2ETT'
+  };
+
+  let updateType = {
+    text: 'E2E Transaction Type updated',
+    description: 'Test transaction type updated',
+    type: 'expense',
+    prefix: 'E2ETT_UPDATED'
+  };
+
+  let otherType = {
+    text: 'E2E Other Transaction Type',
+    description: 'Test Other transaction type',
+    type: 'other',
+    prefix: 'E2EOTT',
+    other: 'OTHER_TYPE'
+  };
+
+  it('Successfully creates a transaction type', () => {
+    FU.buttons.create();
+    FU.input('$ctrl.transactionType.text', newType.text);
+    FU.input('$ctrl.transactionType.description', newType.description);
+    FU.input('$ctrl.transactionType.prefix', newType.prefix);
+    element(by.model('$ctrl.transactionType.type')).click();
+    element(by.css(`[value=${newType.type}]`)).click();
+    FU.buttons.submit();
+    components.notification.hasSuccess();
+  });
+
+  it('Successfully updates an existing transaction type', () => {
+    element(by.css('[data-edit-type="' + identifier + '"]')).click();
```

Comment:
Instead of using the identifier (the user will not select by the identifier), can you use the `text` or another label the user will use to select the correct transaction type?